app/components/contributor-list/component.ts

```diff
@@ -7,35 +7,77 @@ import I18N from 'ember-i18n/services/i18n';
 import Contributor from 'ember-osf-web/models/contributor';
 import defaultTo from 'ember-osf-web/utils/default-to';
 
+export interface Contrib {
+    title: string;
+    id: string | undefined;
+}
+
 @tagName('span')
 export default class ContributorList extends Component {
     @service i18n!: I18N;
-
+    max = 3;
+    nodeId?: string;
+    useLink?: boolean;
     contributors: Contributor[] = defaultTo(this.contributors, A([]));
 
     @computed('contributors.[]')
-    get contributorList(): string {
-        const contributors = this.contributors.toArray();
+    get contributorList(this: ContributorList): Contrib[] {
+        if (!this.get('contributors')) {
+            return [];
+        }
+        const contributors = this.get('contributors').toArray();
 
         if (!(contributors && contributors.length)) {
-            return '';
+            return [];
         }
 
-        const len = contributors.length;
-        const max = 3;
+        const names: Contrib[] = contributors
+            .slice(0, this.get('max'))
+            .map(c => ({
+                title: c.users.get('familyName') || c.users.get('givenName') || c.users.get('fullName'),
+                id: c.users.get('id'),
+            }));
 
-        const names: string[] = contributors
-            .slice(0, max)
-            .map(c => c.users.get('familyName') || c.users.get('givenName') || c.users.get('fullName'));
+        return names;
+    }
 
-        const and = this.i18n.t('general.and');
-        const more = this.i18n.t('general.more');
+    @computed('contributorList')
+    get first(this: ContributorList): Contrib | undefined {
+        return this.get('contributorList').slice(0, 1)[0];
+    }
+
+    @computed('contributorList', 'contributors.[]')
+    get last(this: ContributorList): Contrib | void {
+        const contributors = this.get('contributorList');
 
-        if (len < 3) {
-            return names.join(` ${and} `);
+        if (contributors.length < 2) {
+            return;
         }
 
-        const last = len <= max ? names.splice(-1) : `${len - max} ${more}`;
-        return [...names, `${and} ${last}`].join(', ');
+        const len = this.get('contributors').toArray().length;
+
+        if (len <= this.get('max')) {
+            return contributors.splice(-1)[0];
+        }
+        return {
+            title: `${len - this.get('max')} ${this.get('i18n').t('general.more')}`,
+            id: this.get('nodeId'),
+        };
+    }
+
+    @computed('contributorList', 'contributors.[]')
+    get rest(this: ContributorList): Contrib[] {
+        const contributors = this.get('contributorList');
+
+        if (!(contributors && contributors.length)) {
+            return [];
+        }
+
+        const len = this.get('contributors').toArray().length;
+
+        if (len <= this.get('max')) {
+            return contributors.slice(1, contributors.length - 1);
+        }
+        return contributors.slice(1, this.get('max'));
```

Comment:
Since `contributorsList` is already limited to length `max`, shouldn't `rest` just be
```ts
return this.contributorsList.slice(1, -1);?
```